pallene/checker.lua

```diff
@@ -183,13 +196,16 @@ function Checker:from_ast_type(ast_typ)
             scope_error(ast_typ.loc,  "type '%s' is not declared", name)
         elseif cname._tag == "checker.Name.Type" then
             return cname.typ
-        elseif cname._tag == "checker.Name.Module" and cname.name == "string" then
-            -- Currently the string type appears in the scope as a module because of functions like
-            -- string.char and string.sub. In the future we might want to consider extending this
-            -- feature to other modules that also count as types.
-            return types.T.String()
+        elseif cname._tag == "checker.Name.Module" then
+            if cname.shadowed_type then
+                return cname.shadowed_type.typ
+            else
+                -- fallthrough
+            end
+        else
+            -- fallthrough
         end
-        type_error(ast_typ.loc, "'%s' isn't a type", name)
+        type_error(ast_typ.loc, "'%s' is not a type", name)
```

Comment:
I'm not against this change, but I'm curious as to why it's necessary.

Reply:
The contraction was more informal.